Server/Quality_of_Life_changer/CalendarAdapter.cs

```diff
@@ -1,58 +1,130 @@
-    private readonly CalendarService _calendarService;
+    private static readonly string[] CalendarScopes = {CalendarService.Scope.Calendar};
+
+    private CalendarService _calendarService;
+
+    private ClientSecrets _secrets;
 
     public CalendarAdapter()
     {
-        //UserCredential credential;
-        var credential =
-            GoogleCredential.FromStream(new FileStream("credentials.json", FileMode.Open, FileAccess.Read))
-                .CreateScoped(Scopes);
+        var fileName = GetFileName();
+
+        SetupSecrets(fileName);
+
+        SetupCalendar();
+    }
+
+    public async Task<IEnumerable<CalendarEvent>> GetTodayEvents()
+    {
+        var todayEvents = new List<CalendarEvent>();
+
+        var calendars = await GetCalendarList();
+
+        foreach (var calendar in calendars.Items)
+        {
+            var request = CreateRequest(calendar);
+
+            SetRequestOptions(request);
+
+            var events = await GetEvents(request);
+
+            await AddEventsFromRequestTo(todayEvents, events);
+        }
+
+        return todayEvents;
+    }
+
+    private void SetupSecrets(string fileName)
+    {
+        using var stream = new FileStream(fileName, FileMode.Open, FileAccess.Read);
+        _secrets = Load(stream).Secrets;
+    }
+
+    private static string GetFileName()
+    {
+        return Directory.GetFiles(@".", "client_secret*").First();
+    }
+
+    private async Task SetupCalendar()
+    {
+        var credential = await CreateCredential();
 
-        // Create Google Calendar API service.
+        CreateCalendarService(credential);
+    }
+
+    private async Task<UserCredential> CreateCredential()
+    {
+        return await GoogleWebAuthorizationBroker.AuthorizeAsync(
+            _secrets,
+            CalendarScopes,
+            "user", CancellationToken.None);
+    }
+
+    private async Task<CalendarList> GetCalendarList()
+    {
+        return await _calendarService.CalendarList.List().ExecuteAsync();
+    }
+
+    private void CreateCalendarService(UserCredential credential)
+    {
         _calendarService = new CalendarService(new BaseClientService.Initializer
         {
             HttpClientInitializer = credential,
             ApplicationName = ApplicationName
         });
     }
 
-    public IEnumerable<CalendarEvent> GetTodayEvents()
+    private EventsResource.ListRequest CreateRequest(CalendarListEntry calendar)
     {
-        // Define parameters of request.
-        var request = _calendarService.Events.List("6c73fcbpqb4la5891j62c6l7ndk5na8s@import.calendar.google.com");
-        request.TimeMin = DateTime.Now.AddYears(-1);
-        request.ShowDeleted = false;
-        request.SingleEvents = true;
-        request.MaxResults = 10;
-        request.OrderBy = EventsResource.ListRequest.OrderByEnum.StartTime;
+        return _calendarService.Events.List(calendar.Id);
+    }
+
+    private async Task<Events> GetEvents(EventsResource.ListRequest request)
```

Comment:
there is nothing Events specific in this method, so the naming is wrong, i'd call it `ExecuteRequest`, since it's the only thing this method does.
although, i think this method is redundant. you simply can call `ExecuteAsync()` on request without any wrappers like this method.